scripts/course_ladder.py

```diff
@@ -0,0 +1,39 @@
+# required modules: pymongo, dnspython
+
+from pymongo import MongoClient
+from pymongo.errors import OperationFailure
+
+from state_machine import State, StateMachine, Context
+from states import *
```

Comment:
`from states import Forum`
avoid using `from xxx import *`

Reply:
line 13-14 is setting up all the states, therefore it need to import all the states from states.py. Only importing Forum does not work with the current implementation

I just kinda copy and move code from my 291 mini-project to use it here, we can discuss about better implementations